# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxFT2Utils.cpp
# Commit: cd5ac5f6ff6e
# Full Hash: cd5ac5f6ff6e626bd69e4d42ff9d1d7078abae03
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2022-03-01 09:40:29
# Regressor Bug: 1756720
# File Overlap Count: 1
# Description:
#   Bug 1756720 - When using legacy MS Symbol fonts, map character codes from the U+00xx range to the PUA range U+F0xx found in the cmap subtables. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D139724
# ==============================================================================

diff -r b547c1a2ef61 -r cd5ac5f6ff6e gfx/thebes/gfxFT2Utils.cpp
--- a/gfx/thebes/gfxFT2Utils.cpp	Mon Feb 28 20:34:33 2022 +0000
+++ b/gfx/thebes/gfxFT2Utils.cpp	Mon Feb 28 20:38:38 2022 +0000
@@ -36,10 +36,16 @@
     }
   }
 
-  return FcFreeTypeCharIndex(mFace, aCharCode);
+  uint32_t gid = FcFreeTypeCharIndex(mFace, aCharCode);
 #else
-  return FT_Get_Char_Index(mFace, aCharCode);
+  uint32_t gid = FT_Get_Char_Index(mFace, aCharCode);
 #endif
+  if (!gid && mFace->charmap->encoding == FT_ENCODING_MS_SYMBOL) {
+    if (auto pua = gfxFontUtils::MapLegacySymbolFontCharToPUA(aCharCode)) {
+      gid = FT_Get_Char_Index(mFace, pua);
+    }
+  }
+  return gid;
 }
 
 typedef FT_UInt (*GetCharVariantFunction)(FT_Face face, FT_ULong charcode,